### ISSUE TITLE
Some questions about the codes

### ISSUE THREAD

**Issue by wanginwang** [Goldesel23/Temporal-Ensembling-for-Semi-Supervised-Learning#7]
I'm sorry,..., but your codes seem to have several bugs, and I think it's why it can't reach the accuracy that the paper Temporal Ensembling... reported.
1. you normalize the SVHN images with axes [1,2,3], but they are [32, 32, 3, Num] before you did the reshape option.
2. your model's output is classification possiblility after Softmax activation, but you computed the loss with another rebundant softmax(e.g. tf.losses.softmax_cross_entropy...)
3. your data augmentation is a translation op,
`Args:
translations: A 2-element list representing [dx, dy] or a matrix of 2-element lists representing [dx, dy] to translate for each image (for a batch of images). The rank must be statically known (the shape is not TensorShape(None).
name: The name of the op.`
but you only give an element for each image instead of a 2-element list [dx, dy]...
4.It seems that your weight normalization's reimplementations also have some bugs which I don't know. In addition, I use it with a simple NN mnist supervised classification task, it got a poor test accuracy (53%), which a 97.6% test accuracy can be obtained without using your Weight norm under the same condition.

**Comment by tensorfreitas** [comment]
Hi @wanginwang thanks for taking the time to analyze the code!!!

Regarding your questions:
1. I will analyze this since I wrote it a long time ago, but I think you are correct the normalization should have gone before I swap the axis and after the reshape. 
2. Regarding that I think I have already answered here: #6
3. You are correct, thanks for spotting that. I have to check why it didn't gave an error. Will get back to you! 
4. Can you elaborate a little but more on why do you think it's buggy? I have adapted from the original openai implementation.

**Comment by tensorfreitas** [comment]
Solved the bugs you detected. Thanks @wanginwang ! 

https://github.com/Goldesel23/Temporal-Ensembling-for-Semi-Supervised-Learning/commit/be8ea338692d41beb5132dbb415efd54a0dcd31e
